libcxx/include/__algorithm/iterator_operations.h

```diff
@@ -85,6 +86,56 @@ struct _IterOps<_ClassicAlgPolicy> {
     std::advance(__iter, __count);
   }
 
+  // advance with sentinel, a la std::ranges::advance
+  // it's unclear whether _Iter has a difference_type and whether that's signed, so we play it safe:
```

Comment:
I was trying to preserve compatibility with the previous implementation of `std::set_intersection()`, which didn't make use of `difference_type`. I completely agree with you that the standard requires `LegacyInputIterator`, which mandates the presence of `difference_type`, but it felt like a small price to pay to preserve the flexibility of the previous implementation...

I'm happy to tighten the requirements if you're sure.

Reply:
I think maybe I understand where you're coming from, are you concerned about this not being a faithful version of `std::advance()`, and the similarity confusing others into making mistakes? When I wrote it like that, I was thinking that having a less strict fallback for the function when operating in `_ClassicAlgPolicy` would be a net positive, because it allows us to preserve pre-existing behaviour when using it, but I think I understand your point of view. Personally, I would probably leave it as it is (it's targeted at libc++ implementors and the whole thing will be inlined anyway), but if you disagree, I have two alternatives I'd like to suggest:
1. Rename this to something like `advance_with_custom_difference_type()` so that the intent is explicit, and add a comment explaining that its goal is to avoid introducing or enforcing requirements in legacy code. That way we avoid breaking existing user code without introducing a potential source of confusion.
2. Enforce the `LegacyInputIterator` requirement. If we go down this path, I would like to add a `static_assert()` at the entry point for `std::set_intersection()`, to help users identify the source of the new failure.

What do you think?

Reply:
@ldionne, sorry for nagging, but could you please give feedback on this and on the PR in general?